network.h

```diff
@@ -65,6 +65,9 @@ class network
     ///Mutates the activation of the weights of the network - they get switched on and off
     void mutate_activation(const double &mut_rate, std::mt19937_64 &rng);
 
+    ///changes the pointer to another activation function
+    void  swap_activation_function(std::function<double(double)> new_func) {new_func.swap(m_activation_function);}
```

Comment:
What is this for?

Reply:
to change the activation function of a network on the fly, in case we need to change it.

Reply:
Should this exist before we actually need it?